service_client/metrics.py

```diff
@@ -0,0 +1,53 @@
+"""
+Metrics for the MCP server.
+
+This module provides metrics for the MCP server.
+"""
+
+from typing import Callable, Any
+from functools import wraps
+
+from prometheus_client import (
+    CONTENT_TYPE_LATEST,
+    Counter,
+    Histogram,
+    generate_latest,
+)
+from starlette.responses import PlainTextResponse
+
+
+# Define counter for request count
+REQUEST_COUNT = Counter(
+    "assisted_service_mcp_tool_request_count",
+    "Request count",
+    ["tool"],
+)
+
+# Define histogram for request latency
+REQUEST_LATENCY = Histogram(
+    "assisted_service_mcp_tool_request_duration",
+    "Request latency",
+    ["tool"],
+    buckets=(0.1, 1.0, 10.0, 30.0, float("inf")),
+)
+
+
+def track_tool_usage(tool_name: str) -> Callable:
+    """Decorator to track tool usage metrics"""
+
+    def decorator(func: Callable) -> Callable:
+        @wraps(func)
+        async def wrapper(*args, **kwargs) -> Any:
+            REQUEST_COUNT.labels(tool=tool_name).inc()
```

Comment:
Would it be possible to use `func.__name__` here to avoid repeating the name of the tool in each call to `track_tool_usage`?

Reply:
good idea. implemented in https://github.com/openshift-assisted/assisted-service-mcp/pull/35/commits/fdfba24a7400e5937a8b71a028c165de950b9027